src/core/lib/event_engine/tcp_socket_utils.cc

```diff
@@ -70,6 +72,45 @@ absl::StatusOr<std::string> GetScheme(
           "Unknown scheme: %d", resolved_address.address()->sa_family));
   }
 }
+
+#ifdef GRPC_HAVE_UNIX_SOCKET
+absl::StatusOr<std::string> ResolvedAddrToUriUnixIfPossible(
+    EventEngine::ResolvedAddress* resolved_addr) {
```

Comment:
Is checking `nullptr` necessary?

Reply:
Not really. This function is static to this file and not exposed outside.